Panic/Error When Deleting IstioOperator CRD when Using Revisions

**Bug description**

Istio Operator will panic if I try to delete a control plane that uses revisions.

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**

**Steps to reproduce the bug**

I have an IstioOperator defined as such using Istio Revisions:

```
kubectl apply -f - <<EOF
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: gloo-istiocontrolplane
spec:
  profile: default
  tag: 1.10.2
  revision: 1-10-2
  components:
    pilot:
      k8s:
        resources:
          requests:
            memory: 10Mi
            cpu: 10m
    ingressGateways:
    - name: istio-ingressgateway
      k8s:
        resources:
          requests:
            memory: 10Mi
            cpu: 10m
EOF
```

After applying the CRD, and if I were to delete it as such:

```
k delete istiooperator -n istio-system gloo-istiocontrolplane
```

The Istio control plane will not be removed. Looking in the logs for the Istio Operator, I see this:

```
2021-06-25T05:20:59.923844Z	info	installer	Deleting IstioOperator gloo-istiocontrolplane
2021-06-25T05:20:59.998276Z	error	klog	Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 550 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x2a7a560, 0x5175070)
	k8s.io/apimachinery@v0.20.5/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.20.5/pkg/util/runtime/runtime.go:48 +0x86
panic(0x2a7a560, 0x5175070)
	runtime/panic.go:965 +0x1b9
istio.io/istio/operator/pkg/translate.GetEnabledComponents(0xc000c469c0, 0xc000c0f340, 0x2eaf980, 0x30, 0x1007f5c51f80108, 0x7f5c2af74c20)
	istio.io/istio/operator/pkg/translate/translate_common.go:105 +0x90
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).PruneControlPlaneByRevisionWithController(0xc000c09d60, 0xc000c469c0, 0xc001dc8e70, 0xc001dc8e40, 0x34c41e0)
	istio.io/istio/operator/pkg/helmreconciler/prune.go:111 +0x8a
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).Delete(0xc000c09d60, 0x0, 0x0)
	istio.io/istio/operator/pkg/helmreconciler/reconciler.go:278 +0x74
istio.io/istio/operator/pkg/controller/istiocontrolplane.(*ReconcileIstioOperator).Reconcile(0xc0013ea760, 0x3590428, 0xc001dacc60, 0xc0014b9090, 0xc, 0xc0014518f0, 0x16, 0xc001dacc60, 0xc00004c000, 0x2c84040, ...)
	istio.io/istio/operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go:250 +0x411
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0013f41e0, 0x3590380, 0xc0013e1100, 0x2b91120, 0xc00155dac0)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0013f41e0, 0x3590380, 0xc0013e1100, 0xc000c3fe00)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x3590380, 0xc0013e1100)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc001789750)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc001dc5f50, 0x3512900, 0xc001dacc00, 0xc0013e1101, 0xc0013f85a0)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc001789750, 0x3b9aca00, 0x0, 0xd58701, 0xc0013f85a0)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x3590380, 0xc0013e1100, 0xc001b5fa50, 0x3b9aca00, 0x0, 0x31fb501)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x3590380, 0xc0013e1100, 0xc001b5fa50, 0x3b9aca00)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x40d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2278fb0]

goroutine 550 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.20.5/pkg/util/runtime/runtime.go:55 +0x109
panic(0x2a7a560, 0x5175070)
	runtime/panic.go:965 +0x1b9
istio.io/istio/operator/pkg/translate.GetEnabledComponents(0xc000c469c0, 0xc000c0f340, 0x2eaf980, 0x30, 0x1007f5c51f80108, 0x7f5c2af74c20)
	istio.io/istio/operator/pkg/translate/translate_common.go:105 +0x90
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).PruneControlPlaneByRevisionWithController(0xc000c09d60, 0xc000c469c0, 0xc001dc8e70, 0xc001dc8e40, 0x34c41e0)
	istio.io/istio/operator/pkg/helmreconciler/prune.go:111 +0x8a
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).Delete(0xc000c09d60, 0x0, 0x0)
	istio.io/istio/operator/pkg/helmreconciler/reconciler.go:278 +0x74
istio.io/istio/operator/pkg/controller/istiocontrolplane.(*ReconcileIstioOperator).Reconcile(0xc0013ea760, 0x3590428, 0xc001dacc60, 0xc0014b9090, 0xc, 0xc0014518f0, 0x16, 0xc001dacc60, 0xc00004c000, 0x2c84040, ...)
	istio.io/istio/operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go:250 +0x411
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0013f41e0, 0x3590380, 0xc0013e1100, 0x2b91120, 0xc00155dac0)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0013f41e0, 0x3590380, 0xc0013e1100, 0xc000c3fe00)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x3590380, 0xc0013e1100)
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:185 +0x37
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc001789750)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc001dc5f50, 0x3512900, 0xc001dacc00, 0xc0013e1101, 0xc0013f85a0)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc001789750, 0x3b9aca00, 0x0, 0xd58701, 0xc0013f85a0)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x3590380, 0xc0013e1100, 0xc001b5fa50, 0x3b9aca00, 0x0, 0x31fb501)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:185 +0xa6
k8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x3590380, 0xc0013e1100, 0xc001b5fa50, 0x3b9aca00)
	k8s.io/apimachinery@v0.20.5/pkg/util/wait/wait.go:99 +0x57
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x40d
```

If I use the same IstioOperator CRD without `revision: 1-10-2` I do not run into this issue. It seems to be related to Istio Revisions.

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

Istio: 1.10.2
`image: docker.io/istio/operator:1.10.2`

Client Version: v1.19.7
Server Version: v1.20.2+k3s1

**How was Istio installed?**

I installed the Istio Operator via Helm for 1.10.2

**Environment where the bug was observed (cloud vendor, OS, etc)**

Civo

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

--- a/operator/pkg/translate/translate_common.go
+++ b/operator/pkg/translate/translate_common.go
@@ -102,13 +102,13 @@ func GetEnabledComponents(iopSpec *v1alpha1.IstioOperatorSpec) ([]string, error)
 			}
 		}
 		for _, c := range iopSpec.Components.IngressGateways {
-			if c.Enabled.Value {
+			if c.Enabled.GetValue() {
 				enabledComponents = append(enabledComponents, string(name.IngressComponentName))
 				break
 			}
 		}
 		for _, c := range iopSpec.Components.EgressGateways {
-			if c.Enabled.Value {
+			if c.Enabled.GetValue() {
 				enabledComponents = append(enabledComponents, string(name.EgressComponentName))
 				break
 			}

